x-pack/legacy/plugins/siem/server/lib/detection_engine/alerts/signals_alert_type.ts

```diff
@@ -79,20 +99,120 @@ export const signalsAlertType = ({ logger }: { logger: Logger }): AlertType => {
         references,
       });
 
+      // format scroll search result for signals index.
+      const buildBulkBody = (doc: SearchHit): SignalHit => ({
+        ...doc._source,
+        signal: {
+          rule_revision: 1,
+          rule_id: id,
+          rule_type: type,
+          parent: {
+            id: doc._id,
+            type: 'event',
+            depth: 1,
+          },
+          name,
+          severity,
+          description,
+          time_detected: Date.now(),
+          index_patterns: indexPatterns,
+          references: refs,
+        },
+      });
+
+      // Makes a single bulk index request
+      const singleBulkIndex = async (sr: SearchResponse<object>): Promise<boolean> => {
+        const bulkBody = sr.hits.hits.flatMap((doc: SearchHit) => [
+          {
+            index: {
+              _index: process.env.SIGNALS_INDEX || '.siem-signals-10-01-2019',
+              _id: doc._id,
+            },
+          },
+          buildBulkBody(doc),
+        ]);
+        const firstResult = await services.callCluster('bulk', {
+          refresh: true,
+          body: bulkBody,
+        });
+        if (firstResult.errors) {
+          logger.error(
+            `[-] bulkResponse had errors: ${JSON.stringify(firstResult.errors, null, 2)}}`
+          );
+          return false;
+        }
+        return true;
+      };
```

Comment:
I would move each of these functions to their own file(s) and ensure the functions are pure where we can have them.

We will want to keep the size of `signals_alert_type.ts` as tiny as possible.

Reply:
I don't mind the extra argument passing instead of using closures as it makes things easier to reason about with pure functions and it will make unit testing when we have bugs easier as well.